purchase_order_status/models/purchase.py

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+# © 2019 Houssine BAKKALI, Coop IT Easy SCRLfs
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openerp import api, fields, models
+from openerp.tools.float_utils import float_compare
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    @api.depends('state', 'order_line.qty_invoiced', 'order_line.product_qty')
+    def _get_invoiced(self):
+        precision = self.env['decimal.precision'].precision_get('Product Unit of Measure') #noqa
+        for po in self:
+            invoice_status = 'no'
+            if po.state in ['purchase', 'done']:
+                if any(float_compare(line.qty_invoiced, line.product_qty, precision_digits=precision) == -1 for line in po.order_line): #noqa
+                    invoice_status = 'to invoice'
+                elif all(float_compare(line.qty_invoiced, line.product_qty, precision_digits=precision) >= 0 for line in po.order_line): #noqa
+                    invoice_status = 'invoiced'
+                else:
+                    invoice_status = 'no'

```

Comment:
Nice use of a `any` and `all` but don't abuse the `# noqa` tag, this is more readable with 

```python
if any(
    float_compare(line.qty_invoiced, line.product_qty, precision_digits=precision)
    for line in po.order_line)): 
```
or 
```python
def comp(a, b):
    return float_compare(a, b, precision_digits=precision)
if any(comp(line.qty_invoiced, line.product_qty) == -1 for line in po.order_line): 
    ...
elif all(comp(line.qty_invoiced, line.product_qty) >= 0 for line in po.order_line):
   ...
else:
   ...
```

Reply:
@robinkeunen readability is relative ;)

Reply:
done